src/uiselecttarget.c

```diff
@@ -0,0 +1,483 @@
+        ldr r4, _0804F954  @ sSelectTargetList\n\
+        cmp r2, r1\n\
+        bge _0804F93A\n\
+        adds r5, r6, #0\n\
+        adds r3, r4, #0\n\
+        adds r3, #0xc\n\
+        adds r1, r4, #0\n\
+        subs r1, #0xc\n\
+    _0804F92A:\n\
+        str r1, [r1, #0x14]\n\
+        str r3, [r1, #0x10]\n\
+        adds r3, #0xc\n\
+        adds r1, #0xc\n\
+        adds r2, #1\n\
+        ldr r0, [r5]\n\
+        cmp r2, r0\n\
+        blt _0804F92A\n\
+    _0804F93A:\n\
+        ldr r1, [r6]\n\
+        subs r1, #1\n\
+        lsls r0, r1, #1\n\
+        adds r0, r0, r1\n\
+        lsls r0, r0, #2\n\
+        adds r0, r0, r4\n\
+        str r0, [r4, #8]\n\
+        str r4, [r0, #4]\n\
+        pop {r4, r5, r6}\n\
+        pop {r0}\n\
+        bx r0\n\
+        .align 2, 0\n\
+    _0804F950: .4byte sSelectTargetCount\n\
+    _0804F954: .4byte sSelectTargetList\n\
+        .syntax divided\n\
+    ");
+}
+
+#endif // NONMATCHING
+
+void TargetSelection_GetRealCursorPosition(ProcPtr proc, int* xPos, int* yPos) {
+    *xPos = (((struct SelectTargetProc*)(proc))->currentTarget)->x * 16;
+    *yPos = (((struct SelectTargetProc*)(proc))->currentTarget)->y * 16;
+
+    return;
+}
+
+void TargetSelection_Loop(ProcPtr proc) {
+    int x, y;
+    int r5;
+
+    if ((0x40 & ((struct SelectTargetProc*)(proc))->unk_34) != 0) {
+        TargetSelection_GetRealCursorPosition(proc, &x, &y);
+        DisplayCursor(x, y, 4);
+        return;
+    }
+
+    TargetSelection_HandleMoveInput(proc);
+
+    r5 = TargetSelection_HandleSelectInput(proc);
+
+    if ((2 & r5) != 0) {
+        EndTargetSelection(proc);
+    }
+
+    if ((4 & r5) != 0) {
+        PlaySoundEffect(0x6A);
+    }
+
+    if ((8 & r5) != 0) {
+        PlaySoundEffect(0x6B);
+    }
+
+    if ((0x10 & r5) != 0) {
+        ClearBg0Bg1();
+    }
+
+    if ((0x20 & r5) != 0) {
+        DeleteFaceByIndex(0);
+    }
+
+    if ((1 & r5) == 0) {
+        TargetSelection_GetRealCursorPosition(proc, &x, &y);
+        if (EnsureCameraOntoPosition(proc, x >> 4, y >> 4) != 1) {
+            DisplayCursor(x, y, 2);
+        }
+    }
+
+    return;
+}
+
+ProcPtr NewTargetSelection(const struct SelectInfo* selectInfo) {
+    struct SelectTargetProc* proc;
+
+    AddSkipThread2();
+    proc = Proc_Start(gUnknown_085B655C, PROC_TREE_3);
+
+    proc->unk_34 = 1;
+    proc->selectRoutines = selectInfo;
+    proc->currentTarget = GetFirstTargetPointer();
+    proc->onAPress = 0;
+
+    if (proc->selectRoutines->onInit) {
+        proc->selectRoutines->onInit(proc);
+    }
+
+    if (proc->selectRoutines->onUnk08) {
+        proc->selectRoutines->onUnk08(proc);
+    }
+
+    if (proc->selectRoutines->onSwitchIn) {
+        proc->selectRoutines->onSwitchIn(proc, proc->currentTarget);
+    }
+
+    gKeyStatusPtr->newKeys = 0;
+
+    return proc;
+}
+
+#if NONMATCHING
+
+ProcPtr NewTargetSelection_Specialized(const struct SelectInfo* selectInfo, int(*onSelect)(ProcPtr, struct SelectTarget*)) {
```

Comment:
How is this one not matching? Do you have a scratch or CE link for this one?

Reply:
Aside from the fact that I forgot to change the return type of the function pointer here (fixed in 349ddf5)...  :) 

I could not get it to use the extra register when setting `onAPress`:

```diff
	push {r4, lr}
	adds r4, r1, #0
-	bl NewTargetSelection
-	adds r1, r0, #0
-	str r4, [r1, #0x38]
+ 	bl NewTargetSelection
+	str r4, [r0, #0x38]
	pop {r4}
	pop {r1}
	bx r1
```

Sorry, but I'm not familiar with what you mean by a scratch or CE link! Could you point me to any materials I could use to learn?

Reply:
Ah - joined the Discord server and figured it out in the end. Here's the link: https://cexplore.karathan.at/z/ofW2pR